pkg/capabilities/triggers/cron_trigger.go

```diff
@@ -0,0 +1,287 @@
+	capabilityResponse, err := wrapCronTriggerEvent(triggerEventId, timestampMsIso8601, scheduledExecutionTimeFormatted)
+	if err != nil {
+		cts.lggr.Errorw("error wrapping trigger event", "err", err)
+	}
+
+	select {
+	case cts.jobs[triggerId].ch <- capabilityResponse:
+	default:
+		cts.lggr.Errorw("channel full, dropping event", "eventID", triggerEventId, "triggerID", triggerID, "capabilityResponse", capabilityResponse)
+	}
+}
+
+func (cts *CronTriggerService) UnregisterTrigger(ctx context.Context, req capabilities.CapabilityRequest) error {
+	cts.mu.Lock()
+	defer cts.mu.Unlock()
+
+	config, err := cts.ValidateConfig(req.Config)
+	if err != nil {
+		return err
+	}
+
+	triggerID := getTriggerID(config.TriggerID, req.Metadata.WorkflowID)
+
+	if _, ok := cts.jobs[triggerID]; !ok {
+		return fmt.Errorf("triggerId %s not found", triggerID)
+	}
+
+	trigger, ok := cts.jobs[triggerID]
+	jobId := trigger.job.ID()
+
+	if ok {
+		cts.scheduler.RemoveJob(jobId)
+		close(trigger.ch)
+	}
+	delete(cts.jobs, triggerID)
+	cts.lggr.Debugw("UnregisterTrigger", "triggerId", triggerID, "jobId", jobId)
+	return nil
+}
+
+// Start the service.
+func (cts *CronTriggerService) Start(ctx context.Context) error {
+	cts.wg.Add(1)
+
+	if cts.scheduler == nil {
+		return errors.New("no scheduler initialized")
+	}
+	cts.scheduler.Start()
+
+	cts.lggr.Info(cts.Name() + " started")
+
+	// block until ready to shut down
+	go func() {
+		defer cts.wg.Done()
+		<-cts.stopCh
+	}()
+
+	return nil
+}
+
+// Close stops the Service.
+// Invariants: After this call the Service cannot be started
+// again, new Service will need to be built to do so.
+func (cts *CronTriggerService) Close() error {
+	err := cts.scheduler.Shutdown()
+	if err != nil {
+		return fmt.Errorf("scheduler shutdown encountered a problem: %s", err)
+	}
+
+	close(cts.stopCh)
+
+	cts.wg.Wait()
+
+	cts.lggr.Info(cts.Name() + " closed")
+
+	return nil
+}
+
+func (cts *CronTriggerService) Ready() error {
+	if cts.scheduler == nil {
+		return errors.New("no scheduler initialized")
+	}
+	return nil
+}
+
+func (cts *CronTriggerService) HealthReport() map[string]error {
+	return map[string]error{cts.Name(): nil}
+}
+
+func (cts *CronTriggerService) Name() string {
+	return "CronTriggerService"
+}
+
+func getTriggerID(triggerID string, wid string) string {
+	return wid + "|" + triggerID
+}
+
+func wrapCronTriggerEvent(eventID string, timestamp string, scheduledExecutionTime string) (capabilities.CapabilityResponse, error) {
+	payload, err := values.Wrap(CronTriggerPayload{ScheduledExecutionTime: scheduledExecutionTime})
+	if err != nil {
+		return capabilities.CapabilityResponse{}, err
+	}
+
+	metadata, err := values.Wrap(nil)
+	if err != nil {
+		return capabilities.CapabilityResponse{}, err
+	}
+
+	triggerEvent := capabilities.TriggerEvent{
+		TriggerType: cronTriggerID,
+		ID:          eventID,
+		Timestamp:   timestamp,
+		Metadata:    metadata,
+		Payload:     payload,
+	}
+
+	eventVal, err := values.Wrap(triggerEvent)
```

Comment:
WrapMap will return a map or error, this would avoid the need to coerce on l 278

Reply:
Good tip!